### ISSUE TITLE
Recreating results

### ISSUE THREAD

**Issue by Fergus-Hayes** [takh04/QCNN#5]
On running `result.py` I do not seem to get a satisfactory accuracy (as well as a rather constant cost) when run with the default settings. Nor if I change the settings. For example, if I change the settings in `result.py` to:

```
Unitaries = ['U_SU4']
U_num_params = [51]
Encodings = ['resize256']
dataset = 'mnist'
classes = [0,1]
binary = False
cost_fn = 'cross_entropy'
``` 

I get back:

```
iteration:  0  cost:  18.608716944569423
iteration:  10  cost:  17.859785037299087
iteration:  20  cost:  16.864074122595234
iteration:  30  cost:  19.85744029015193
iteration:  40  cost:  17.39387649763763
iteration:  50  cost:  15.653618570682836
iteration:  60  cost:  15.863246298781105
iteration:  70  cost:  16.058333044866135
iteration:  80  cost:  16.920172385993272
iteration:  90  cost:  16.298259957299084
iteration:  100  cost:  14.757749928867613
iteration:  110  cost:  17.26280635267795
iteration:  120  cost:  17.245992599610698
iteration:  130  cost:  17.736104607235028
iteration:  140  cost:  18.5683812175255
iteration:  150  cost:  18.09948105855172
iteration:  160  cost:  17.785045925879157
iteration:  170  cost:  19.03336409702668
iteration:  180  cost:  14.876061801082823
iteration:  190  cost:  17.38155438828209
Accuracy for U_SU4 resize256 :0.5366430260047281
```
While I am not sure exactly what the input settings are specifying, from what I understand is that 'resize256' assigns amplitude encoding, and 'U_SU4' assigns the Ansatz 9. However, an accuracy of 54% does not reflect any of the amplitude encoding results shown in Table 1. I notice the cost from the cross entropy is greater than 1 in my output here, while it is less than 1 in Figure 6. Is there something misspecified? How can I recreate your results?

**Comment by takh04** [comment]
Hi. Thanks for the comment. Can you try running the code with "U_num_params = [51]" and show me how it works? "U_num_params" should be list of number of parameters per single ansatz / convolutional layer, which is 15 for U_SU4.

Sorry that I can't provide detailed guides or solutions about the code. I am currently quite busy preparing for my final exam.  I will definitely tidy up the codes and write detailed guides on how to execute the code soon.

**Comment by Fergus-Hayes** [comment]
Thanks for your quick response, and good luck with your exams. I guess you mean setting `U_num_params = [15]`? When I do: 
```
Unitaries = ['U_SU4']
U_num_params = [15]
Encodings = ['resize256']
dataset = 'mnist'
classes = [0,1]
binary = False
cost_fn = 'cross_entropy'
```
I get the result:
```
iteration:  0  cost:  19.257125252816937
iteration:  10  cost:  20.460772978497438
iteration:  20  cost:  19.73641665197323
iteration:  30  cost:  21.313320494553974
iteration:  40  cost:  19.843955777764965
iteration:  50  cost:  20.53238842969016
iteration:  60  cost:  20.173287734461326
iteration:  70  cost:  19.84076700946766
iteration:  80  cost:  20.470032401572343
iteration:  90  cost:  19.873266980045194
iteration:  100  cost:  20.23456204120762
iteration:  110  cost:  20.010674007578658
iteration:  120  cost:  19.296635963292353
iteration:  130  cost:  19.783822575735073
iteration:  140  cost:  19.582073892360587
iteration:  150  cost:  21.249260552245747
iteration:  160  cost:  21.076016066387314
iteration:  170  cost:  19.666745778195118
iteration:  180  cost:  19.562385745153637
iteration:  190  cost:  19.27453822136908
Accuracy for U_SU4 resize256 :0.2723404255319149
```

**Comment by takh04** [comment]
Thanks for letting me know. Surely, the results seem to be problematic. I will definitely have a look at the code and fix it in the near future (in a few weeks). Thanks!

**Comment by takh04** [comment]
Hi, it seems like the parameters are not being trained after the cost function is measured. You can easily fix this by modifying **Training.py**. 
First, change `import numpy as np` to `from pennylane import numpy as np`.
Second, change `params = np.random.randn(total_params)` to `params = np.random.randn(total_params, requires_grad=True)`. I believe you will get results comparable to the paper with these simple results.

There were few errors in the process of updating the codes on Github. I will fix it and update the new codes in few days.

**Comment by Fergus-Hayes** [comment]
Thanks, this does seem to solve the issue.
